path.go

```diff
@@ -172,11 +172,19 @@ func SplitAbsPath(fpath Path) (cid.Cid, []string, error) {
 		return cid.Cid{}, nil, &pathError{error: fmt.Errorf("empty"), path: string(fpath)}
 	}
 
-	c, err := cid.Decode(parts[0])
+	c, err := decodeCid(parts[0])
 	// first element in the path is a cid
 	if err != nil {
 		return cid.Cid{}, nil, &pathError{error: fmt.Errorf("invalid CID: %s", err), path: string(fpath)}
 	}
 
 	return c, parts[1:], nil
 }
+
+func decodeCid(cstr string) (cid.Cid, error) {
+	c, err := cid.Decode(cstr)
+	if err != nil && len(cstr) == 46 && cstr[:2] == "qm" { // https://github.com/ipfs/go-ipfs/issues/7792
+		return cid.Cid{}, fmt.Errorf("%v: This looks like a CIDv0 that has been lowercased. Convert the CIDv0 to CIDv1 in case-insensitive base32 using 'ipfs cid base32 <Qm..>' and try again.", err)
```

Comment:
I would still like this to be much shorter. Can we just say "this looks like a lowercase CID"? Telling the user what to do is a bit out of scope for an error.

Reply:
'This looks like a CIDv0 that has been lowercased; you may wish to convert to case-agnostic CIDv1' or similar? Intent would be not to tell user what to do, but hint that there's a workaround if preserving case isn't an option.